providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -462,8 +524,16 @@ static void *ml_kem_gen(void *vgctx, OSSL_CALLBACK *osslcb, void *cbarg)
         OPENSSL_cleanse(seed, ML_KEM_SEED_BYTES);
     gctx->seed = NULL;
 
-    if (genok)
+    if (genok) {
+#ifdef FIPS_MODULE
+        if (!ml_kem_pairwise_test(key)) {
+            ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT);
+            ossl_ml_kem_key_free(key);
+            return NULL;
+        }
+#endif  /* FIPS_MODULE */
```

Comment:
There should be a similar PCT when a private key is loaded in `fromdata`. FIPS 140-3:

```
The PCT shall be performed either when keys are generated/imported,
prior to the first exportation, or prior to the first operational use
(if not exported before the first use).
```